File: eFormAPI/Plugins/InstallationChecking.Pn/InstallationChecking.Pn/Infrastructure/Models/InstallationsRequestModel.cs
using Microting.InstallationCheckingBase.Infrastructure.Enums;

namespace InstallationChecking.Pn.Infrastructure.Models
{
    using Microting.eFormApi.BasePn.Infrastructure.Models.Common;

    public class InstallationsRequestModel : PaginationModel
    {
        public string NameFilter { get; set; }

        public InstallationType? Type { get; set; }
            = InstallationType.Installation;

        public InstallationState? State { get; set; }

        public string Sort { get; set; }

        public bool IsSortDsc { get; set; }
    }
}

File: eform-client/src/app/plugins/modules/installationchecking-pn/models/installations-request.model.ts
import { InstallationStateEnum, InstallationTypeEnum } from '../const/enums';

export class InstallationsRequestModel {
  nameFilter: string;
  type: InstallationTypeEnum;
  state: InstallationStateEnum;

  sort: string;
  offset: number;
  pageSize: number;
  isSortDsc: boolean;

  constructor() {
    this.sort = 'Id';
    this.isSortDsc = true;
    this.pageSize = 10;
    this.offset = 0;
  }
}
